crates/gateway-config/src/mcp.rs

```diff
@@ -0,0 +1,27 @@
+use serde_dynamic_string::DynamicString;
+
+#[derive(Clone, Debug, serde::Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct ModelControlProtocolConfig {
+    /// Whether the MCP service is enabled.
+    #[serde(default)]
+    pub enabled: bool,
+    /// The name of the service.
+    pub name: String,
+    /// The instructions for the LLM how to use the service.
+    pub instructions: Option<String>,
+    /// The service path in the gateway.
+    #[serde(default = "default_mcp_path")]
+    pub path: String,
+    /// Whether mutations are enabled for the MCP service.
+    #[serde(default)]
+    pub enable_mutations: bool,
+    /// Pass the downstream auth header here until the MCP service
+    /// supports passing headers.
+    #[serde(default)]
+    pub authentication: Option<DynamicString<String>>,
```

Comment:
We have header rules that can insert headers, seems clearer and more versatile than this option IMHO.

Reply:
you're also literrally using a `forward` rule when you could just use an `insert` rule, I wonder who wrote the header rules :D /troll

Reply:
This is not how it's going to be done eventually. We will use the header rules, but right now the rmcp crate or any other mcp library cannot pass headers. It's coming, so we can keep this ugly hack until we actually can transfer headers.

Reply:
What I'm suggesting is that you can simply remove this hack and use an `insert` rule today, instead of using both a `forward` rule AND this config option